### PR TITLE
refactor: Convert AbstractFlashcardViewerCommandTest.java to Kotlin

### PR DESCRIPTION
**Pull request by codingtosh**
refactor: Convert AbstractFlashcardViewerCommandTest.java to Kotlin
com.ichi2.anki.AbstractFlashcardViewerCommandTest

Fixes #10783

### CODE REVIEW

File: AnkiDroid/src/test/java/com/ichi2/anki/AbstractFlashcardViewerCommandTest.kt [hunk 1]
```diff
@@ -13,233 +13,181 @@
...
-
-        viewer.executeCommand(ViewerCommand.COMMAND_UNSET_FLAG);
-        viewer.executeCommand(ViewerCommand.COMMAND_UNSET_FLAG);
-
-        assertThat(viewer.getLastFlag(), is(FLAG_NONE));
+    fun doubleNoneDoesNothing() {
+        val viewer = viewer
+        viewer.executeCommand(ViewerCommand.COMMAND_UNSET_FLAG)
+        viewer.executeCommand(ViewerCommand.COMMAND_UNSET_FLAG)
+        MatcherAssert.assertThat(viewer.lastFlag, Matchers.`is`(CardMarker.FLAG_NONE))
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/anki/AbstractFlashcardViewerCommandTest.kt:54]: import : `CardMarker`

File: AnkiDroid/src/test/java/com/ichi2/anki/AbstractFlashcardViewerCommandTest.kt [hunk 2]
```diff
@@ -13,233 +13,181 @@
...
+        viewer.executeCommand(command)
+        viewer.executeCommand(command)
+        MatcherAssert.assertThat(command.toString(), viewer.lastFlag, Matchers.`is`(CardMarker.FLAG_NONE))
+    }
+
+    private val viewer: CommandTestCardViewer
+        get() = CommandTestCardViewer(cardWith(CardMarker.FLAG_NONE))
+
+    private fun cardWith(@Suppress("SameParameterValue") @FlagDef flag: Int): Card {
+        val c = Mockito.mock(Card::class.java)
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/anki/AbstractFlashcardViewerCommandTest.kt:144]: import `mockito`

File: AnkiDroid/src/test/java/com/ichi2/anki/AbstractFlashcardViewerCommandTest.kt [hunk 3]
```diff
@@ -13,233 +13,181 @@
...
+        val flags = intArrayOf(flag)
+        Mockito.`when`(c.userFlag()).then { flags[0] }
+        Mockito.doAnswer { invocation: InvocationOnMock ->
+            flags[0] = invocation.getArgument(0)
+            null
+        }.`when`(c).setUserFlag(ArgumentMatchers.anyInt())
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/anki/AbstractFlashcardViewerCommandTest.kt:150]: import `argumentmatchers`

File: AnkiDroid/src/test/java/com/ichi2/anki/AbstractFlashcardViewerCommandTest.kt [hunk 4]
```diff
@@ -13,233 +13,181 @@
...
+import org.hamcrest.MatcherAssert
+import org.hamcrest.Matchers
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.ArgumentMatchers
+import org.mockito.Mockito
+import org.mockito.invocation.InvocationOnMock
+
+@RunWith(AndroidJUnit4::class)
+class AbstractFlashcardViewerCommandTest : RobolectricTest() {
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/anki/AbstractFlashcardViewerCommandTest.kt:33]: cleanup: `is -> equalTo`
cleanup: `when -> whenever`

**Review by david-allison — CHANGES_REQUESTED** [review]

**Review by david-allison — APPROVED** [review]

**Review by mikehardy — APPROVED** [review]
:fist_right: :fist_left:

**Pull request merged**